tests/testStatsdClient.cpp

```diff
@@ -1,65 +1,154 @@
+void mock(StatsdServer& server, std::vector<std::string>& messages) {
+    do {
+        // Grab the messages that are waiting
+        auto recvd = server.receive();
 
-    // set a new config with a different port
-    client.setConfig("localhost", 8125, "myPrefix.");
-    if (!client.errorMessage().empty()) {
-        std::cerr << client.errorMessage() << std::endl;
-        return EXIT_FAILURE;
-    }
+        // Split the messages on '\n'
+        std::string::size_type start = -1;
+        do {
+            // Keep this message
+            auto end = recvd.find('\n', ++start);
+            messages.emplace_back(recvd.substr(start, end));
+            start = end;
 
-    // Increment "coco"
-    client.increment("coco");
-    if (!client.errorMessage().empty()) {
-        std::cerr << client.errorMessage() << std::endl;
-        return EXIT_FAILURE;
-    }
+            // Bail if we found the special quit message
+            if (messages.back().find("DONE") != std::string::npos) {
+                messages.pop_back();
+                return;
+            }
+        } while (start != std::string::npos);
+    } while (server.errorMessage().empty() && !messages.back().empty());
+}
 
-    // Decrement "kiki"
-    client.decrement("kiki");
+//TODO: would it be ok if we change the API of the client to throw on error instead of having to check a string?
+void throwOnError(const StatsdClient& client) {
     if (!client.errorMessage().empty()) {
         std::cerr << client.errorMessage() << std::endl;
-        return EXIT_FAILURE;
+        throw std::runtime_error(client.errorMessage());
     }
+}
 
-    // Adjusts "toto" by +3
-    client.count("toto", 2, 0.1f);
-    if (!client.errorMessage().empty()) {
-        std::cerr << client.errorMessage() << std::endl;
-        return EXIT_FAILURE;
+void testErrorConditions() {
+    // Connect to a rubbish ip and make sure initialization failed
+    StatsdClient client{"256.256.256.256", 8125, "myPrefix.", 20};
+    if (client.errorMessage().empty()) {
+        std::cerr << "Should not be able to connect to ridiculous ip" << std::endl;
+        throw std::runtime_error("Should not be able to connect to ridiculous ip");
     }
+}
 
-    // Record a gauge "titi" to 3
-    client.gauge("titi", 3);
-    if (!client.errorMessage().empty()) {
-        std::cerr << client.errorMessage() << std::endl;
-        return EXIT_FAILURE;
-    }
+void testReconfigure() {
+    StatsdServer server;
 
-    // Record a timing of 2ms for "myTiming"
-    client.timing("myTiming", 2, 0.1f);
-    if (!client.errorMessage().empty()) {
-        std::cerr << client.errorMessage() << std::endl;
-        return EXIT_FAILURE;
+    StatsdClient client("localhost", 8125, "first.");
+    client.send("foo", 1, MetricType::COUNTER, 1.f);
+    if(server.receive() != "first.foo:1|c")
+        throw std::runtime_error("Incorrect stat received");
+
```

Comment:
i dont know if there is a style rule to force the braces, but it seems the current style doesnt do that. for me it seems to, if it could fit it, put it on one line. i figure you'd rather update the style in another pr to handle adding braces anyway though